### DISCUSSION TITLE
Rails - Can't run migrations against my Database

### DISCUSSION THREAD

**Discussion by arsalan-menhaj** [supabase/supabase#21681] [Questions]
Hi all!
I was following the guide for [Supabase with Rails](https://supabase.com/docs/guides/getting-started/quickstarts/ruby-on-rails), but after adding the DB url as an environment variable and generating the first model, I'm unable to run a migration (step 3). I've ensured that the DB url contains the current password, but I'm still having this issue. Below is the output trace:

** Invoke db:migrate (first_time)
** Invoke db:load_config (first_time)
** Invoke environment (first_time)
** Execute environment
** Execute db:load_config
** Execute db:migrate
bin/rails aborted!
ActiveRecord::ConnectionNotEstablished: connection to server on socket "/var/run/postgresql/.s.PGSQL.5432" failed: No such file or directory
        Is the server running locally and accepting connections on that socket?
/var/lib/gems/3.1.0/gems/activerecord-7.1.3.2/lib/active_record/connection_adapters/postgresql_adapter.rb:80:in `rescue in new_client'
/var/lib/gems/3.1.0/gems/activerecord-7.1.3.2/lib/active_record/connection_adapters/postgresql_adapter.rb:68:in `new_client'
/var/lib/gems/3.1.0/gems/activerecord-7.1.3.2/lib/active_record/connection_adapters/postgresql_adapter.rb:980:in `connect'
/var/lib/gems/3.1.0/gems/activerecord-7.1.3.2/lib/active_record/connection_adapters/postgresql_adapter.rb:992:in `reconnect'
/var/lib/gems/3.1.0/gems/activerecord-7.1.3.2/lib/active_record/connection_adapters/abstract_adapter.rb:685:in `block in reconnect!'
/var/lib/gems/3.1.0/gems/activesupport-7.1.3.2/lib/active_support/concurrency/null_lock.rb:9:in `synchronize'
/var/lib/gems/3.1.0/gems/activerecord-7.1.3.2/lib/active_record/connection_adapters/abstract_adapter.rb:684:in `reconnect!'
/var/lib/gems/3.1.0/gems/activerecord-7.1.3.2/lib/active_record/connection_adapters/abstract_adapter.rb:788:in `verify!'
/var/lib/gems/3.1.0/gems/activerecord-7.1.3.2/lib/active_record/connection_adapters/abstract_adapter.rb:795:in `connect!'
/var/lib/gems/3.1.0/gems/activerecord-7.1.3.2/lib/active_record/connection_adapters/abstract_adapter.rb:1001:in `block in with_raw_connection'
/var/lib/gems/3.1.0/gems/activesupport-7.1.3.2/lib/active_support/concurrency/null_lock.rb:9:in `synchronize'
/var/lib/gems/3.1.0/gems/activerecord-7.1.3.2/lib/active_record/connection_adapters/abstract_adapter.rb:1000:in `with_raw_connection'
/var/lib/gems/3.1.0/gems/activerecord-7.1.3.2/lib/active_record/connection_adapters/abstract_adapter.rb:1113:in `valid_raw_connection'
/var/lib/gems/3.1.0/gems/activerecord-7.1.3.2/lib/active_record/connection_adapters/postgresql_adapter.rb:616:in `get_database_version'
/var/lib/gems/3.1.0/gems/activerecord-7.1.3.2/lib/active_record/connection_adapters/schema_cache.rb:374:in `database_version'
/var/lib/gems/3.1.0/gems/activerecord-7.1.3.2/lib/active_record/connection_adapters/schema_cache.rb:70:in `database_version'
/var/lib/gems/3.1.0/gems/activerecord-7.1.3.2/lib/active_record/connection_adapters/schema_cache.rb:200:in `database_version'
/var/lib/gems/3.1.0/gems/activerecord-7.1.3.2/lib/active_record/connection_adapters/abstract_adapter.rb:871:in `database_version'
/var/lib/gems/3.1.0/gems/activerecord-7.1.3.2/lib/active_record/connection_adapters/postgresql_adapter.rb:644:in `check_version'
/var/lib/gems/3.1.0/gems/activerecord-7.1.3.2/lib/active_record/connection_adapters/abstract/connection_pool.rb:678:in `new_connection'
/var/lib/gems/3.1.0/gems/activerecord-7.1.3.2/lib/active_record/connection_adapters/abstract/connection_pool.rb:723:in `checkout_new_connection'
/var/lib/gems/3.1.0/gems/activerecord-7.1.3.2/lib/active_record/connection_adapters/abstract/connection_pool.rb:702:in `try_to_checkout_new_connection'
/var/lib/gems/3.1.0/gems/activerecord-7.1.3.2/lib/active_record/connection_adapters/abstract/connection_pool.rb:654:in `acquire_connection'
/var/lib/gems/3.1.0/gems/activerecord-7.1.3.2/lib/active_record/connection_adapters/abstract/connection_pool.rb:353:in `checkout'
/var/lib/gems/3.1.0/gems/activerecord-7.1.3.2/lib/active_record/connection_adapters/abstract/connection_pool.rb:182:in `connection'
/var/lib/gems/3.1.0/gems/activerecord-7.1.3.2/lib/active_record/connection_adapters/abstract/connection_handler.rb:246:in `retrieve_connection'
/var/lib/gems/3.1.0/gems/activerecord-7.1.3.2/lib/active_record/connection_handling.rb:287:in `retrieve_connection'
/var/lib/gems/3.1.0/gems/activerecord-7.1.3.2/lib/active_record/connection_handling.rb:254:in `connection'
/var/lib/gems/3.1.0/gems/activerecord-7.1.3.2/lib/active_record/tasks/database_tasks.rb:510:in `migration_connection'/var/lib/gems/3.1.0/gems/activerecord-7.1.3.2/lib/active_record/tasks/database_tasks.rb:243:in `migrate'
/var/lib/gems/3.1.0/gems/activerecord-7.1.3.2/lib/active_record/railties/databases.rake:93:in `block (2 levels) in <main>'
/var/lib/gems/3.1.0/gems/rake-13.1.0/lib/rake/task.rb:281:in `block in execute'
/var/lib/gems/3.1.0/gems/rake-13.1.0/lib/rake/task.rb:281:in `each'
/var/lib/gems/3.1.0/gems/rake-13.1.0/lib/rake/task.rb:281:in `execute'
/var/lib/gems/3.1.0/gems/rake-13.1.0/lib/rake/task.rb:219:in `block in invoke_with_call_chain'
/var/lib/gems/3.1.0/gems/rake-13.1.0/lib/rake/task.rb:199:in `synchronize'
/var/lib/gems/3.1.0/gems/rake-13.1.0/lib/rake/task.rb:199:in `invoke_with_call_chain'
/var/lib/gems/3.1.0/gems/rake-13.1.0/lib/rake/task.rb:188:in `invoke'
/var/lib/gems/3.1.0/gems/rake-13.1.0/lib/rake/application.rb:182:in `invoke_task'
/var/lib/gems/3.1.0/gems/rake-13.1.0/lib/rake/application.rb:138:in `block (2 levels) in top_level'
/var/lib/gems/3.1.0/gems/rake-13.1.0/lib/rake/application.rb:138:in `each'
/var/lib/gems/3.1.0/gems/rake-13.1.0/lib/rake/application.rb:138:in `block in top_level'
/var/lib/gems/3.1.0/gems/rake-13.1.0/lib/rake/application.rb:147:in `run_with_threads'
/var/lib/gems/3.1.0/gems/rake-13.1.0/lib/rake/application.rb:132:in `top_level'
/var/lib/gems/3.1.0/gems/railties-7.1.3.2/lib/rails/commands/rake/rake_command.rb:27:in `block (2 levels) in perform'/var/lib/gems/3.1.0/gems/rake-13.1.0/lib/rake/application.rb:208:in `standard_exception_handling'
/var/lib/gems/3.1.0/gems/railties-7.1.3.2/lib/rails/commands/rake/rake_command.rb:27:in `block in perform'
/var/lib/gems/3.1.0/gems/railties-7.1.3.2/lib/rails/commands/rake/rake_command.rb:44:in `block in with_rake'
/var/lib/gems/3.1.0/gems/rake-13.1.0/lib/rake/rake_module.rb:59:in `with_application'
/var/lib/gems/3.1.0/gems/railties-7.1.3.2/lib/rails/commands/rake/rake_command.rb:41:in `with_rake'
/var/lib/gems/3.1.0/gems/railties-7.1.3.2/lib/rails/commands/rake/rake_command.rb:20:in `perform'
/var/lib/gems/3.1.0/gems/railties-7.1.3.2/lib/rails/command.rb:156:in `invoke_rake'
/var/lib/gems/3.1.0/gems/railties-7.1.3.2/lib/rails/command.rb:73:in `block in invoke'
/var/lib/gems/3.1.0/gems/railties-7.1.3.2/lib/rails/command.rb:149:in `with_argv'
/var/lib/gems/3.1.0/gems/railties-7.1.3.2/lib/rails/command.rb:69:in `invoke'
/var/lib/gems/3.1.0/gems/railties-7.1.3.2/lib/rails/commands.rb:18:in `<main>'
/var/lib/gems/3.1.0/gems/bootsnap-1.18.3/lib/bootsnap/load_path_cache/core_ext/kernel_require.rb:30:in `require'
/var/lib/gems/3.1.0/gems/bootsnap-1.18.3/lib/bootsnap/load_path_cache/core_ext/kernel_require.rb:30:in `require'
bin/rails:4:in `<main>'

Caused by:
PG::ConnectionBad: connection to server on socket "/var/run/postgresql/.s.PGSQL.5432" failed: No such file or directory
        Is the server running locally and accepting connections on that socket?
/var/lib/gems/3.1.0/gems/pg-1.5.5/lib/pg/connection.rb:819:in `connect_start'
/var/lib/gems/3.1.0/gems/pg-1.5.5/lib/pg/connection.rb:819:in `connect_to_hosts'
/var/lib/gems/3.1.0/gems/pg-1.5.5/lib/pg/connection.rb:759:in `new'
/var/lib/gems/3.1.0/gems/pg-1.5.5/lib/pg.rb:63:in `connect'
/var/lib/gems/3.1.0/gems/activerecord-7.1.3.2/lib/active_record/connection_adapters/postgresql_adapter.rb:69:in `new_client'
/var/lib/gems/3.1.0/gems/activerecord-7.1.3.2/lib/active_record/connection_adapters/postgresql_adapter.rb:980:in `connect'
/var/lib/gems/3.1.0/gems/activerecord-7.1.3.2/lib/active_record/connection_adapters/postgresql_adapter.rb:992:in `reconnect'
/var/lib/gems/3.1.0/gems/activerecord-7.1.3.2/lib/active_record/connection_adapters/abstract_adapter.rb:685:in `block in reconnect!'
/var/lib/gems/3.1.0/gems/activesupport-7.1.3.2/lib/active_support/concurrency/null_lock.rb:9:in `synchronize'
/var/lib/gems/3.1.0/gems/activerecord-7.1.3.2/lib/active_record/connection_adapters/abstract_adapter.rb:684:in `reconnect!'
/var/lib/gems/3.1.0/gems/activerecord-7.1.3.2/lib/active_record/connection_adapters/abstract_adapter.rb:788:in `verify!'
/var/lib/gems/3.1.0/gems/activerecord-7.1.3.2/lib/active_record/connection_adapters/abstract_adapter.rb:795:in `connect!'
/var/lib/gems/3.1.0/gems/activerecord-7.1.3.2/lib/active_record/connection_adapters/abstract_adapter.rb:1001:in `block in with_raw_connection'
/var/lib/gems/3.1.0/gems/activesupport-7.1.3.2/lib/active_support/concurrency/null_lock.rb:9:in `synchronize'
/var/lib/gems/3.1.0/gems/activerecord-7.1.3.2/lib/active_record/connection_adapters/abstract_adapter.rb:1000:in `with_raw_connection'
/var/lib/gems/3.1.0/gems/activerecord-7.1.3.2/lib/active_record/connection_adapters/abstract_adapter.rb:1113:in `valid_raw_connection'
/var/lib/gems/3.1.0/gems/activerecord-7.1.3.2/lib/active_record/connection_adapters/postgresql_adapter.rb:616:in `get_database_version'
/var/lib/gems/3.1.0/gems/activerecord-7.1.3.2/lib/active_record/connection_adapters/schema_cache.rb:374:in `database_version'
/var/lib/gems/3.1.0/gems/activerecord-7.1.3.2/lib/active_record/connection_adapters/schema_cache.rb:70:in `database_version'
/var/lib/gems/3.1.0/gems/activerecord-7.1.3.2/lib/active_record/connection_adapters/schema_cache.rb:200:in `database_version'
/var/lib/gems/3.1.0/gems/activerecord-7.1.3.2/lib/active_record/connection_adapters/abstract_adapter.rb:871:in `database_version'
/var/lib/gems/3.1.0/gems/activerecord-7.1.3.2/lib/active_record/connection_adapters/postgresql_adapter.rb:644:in `check_version'
/var/lib/gems/3.1.0/gems/activerecord-7.1.3.2/lib/active_record/connection_adapters/abstract/connection_pool.rb:678:in `new_connection'
/var/lib/gems/3.1.0/gems/activerecord-7.1.3.2/lib/active_record/connection_adapters/abstract/connection_pool.rb:723:in `checkout_new_connection'
/var/lib/gems/3.1.0/gems/activerecord-7.1.3.2/lib/active_record/connection_adapters/abstract/connection_pool.rb:702:in `try_to_checkout_new_connection'
/var/lib/gems/3.1.0/gems/activerecord-7.1.3.2/lib/active_record/connection_adapters/abstract/connection_pool.rb:654:in `acquire_connection'
/var/lib/gems/3.1.0/gems/activerecord-7.1.3.2/lib/active_record/connection_adapters/abstract/connection_pool.rb:353:in `checkout'
/var/lib/gems/3.1.0/gems/activerecord-7.1.3.2/lib/active_record/connection_adapters/abstract/connection_pool.rb:182:in `connection'
/var/lib/gems/3.1.0/gems/activerecord-7.1.3.2/lib/active_record/connection_adapters/abstract/connection_handler.rb:246:in `retrieve_connection'
/var/lib/gems/3.1.0/gems/activerecord-7.1.3.2/lib/active_record/connection_handling.rb:287:in `retrieve_connection'
/var/lib/gems/3.1.0/gems/activerecord-7.1.3.2/lib/active_record/connection_handling.rb:254:in `connection'
/var/lib/gems/3.1.0/gems/activerecord-7.1.3.2/lib/active_record/tasks/database_tasks.rb:510:in `migration_connection'/var/lib/gems/3.1.0/gems/activerecord-7.1.3.2/lib/active_record/tasks/database_tasks.rb:243:in `migrate'
/var/lib/gems/3.1.0/gems/activerecord-7.1.3.2/lib/active_record/railties/databases.rake:93:in `block (2 levels) in <main>'
/var/lib/gems/3.1.0/gems/rake-13.1.0/lib/rake/task.rb:281:in `block in execute'
/var/lib/gems/3.1.0/gems/rake-13.1.0/lib/rake/task.rb:281:in `each'
/var/lib/gems/3.1.0/gems/rake-13.1.0/lib/rake/task.rb:281:in `execute'
/var/lib/gems/3.1.0/gems/rake-13.1.0/lib/rake/task.rb:219:in `block in invoke_with_call_chain'
/var/lib/gems/3.1.0/gems/rake-13.1.0/lib/rake/task.rb:199:in `synchronize'
/var/lib/gems/3.1.0/gems/rake-13.1.0/lib/rake/task.rb:199:in `invoke_with_call_chain'
/var/lib/gems/3.1.0/gems/rake-13.1.0/lib/rake/task.rb:188:in `invoke'
/var/lib/gems/3.1.0/gems/rake-13.1.0/lib/rake/application.rb:182:in `invoke_task'
/var/lib/gems/3.1.0/gems/rake-13.1.0/lib/rake/application.rb:138:in `block (2 levels) in top_level'
/var/lib/gems/3.1.0/gems/rake-13.1.0/lib/rake/application.rb:138:in `each'
/var/lib/gems/3.1.0/gems/rake-13.1.0/lib/rake/application.rb:138:in `block in top_level'

I would appreciate any help. Thanks  in advance!

**Comment by GaryAustin1** [comment]
You might show your connection info with your password hidden.  Ipv4 networks need to use the Supavisor pooling connection and ipv6 can still use the direct connection port.  The Dashboard Database settings has the options.

Note, I don't know for sure if your error is saying it can't connect as I don't use Rails, but seems that way.
